tutorials/Multi_objective_multi_fidelity_BO.ipynb

```diff
@@ -0,0 +1,1720 @@
+    "        model,\n",
+    "        train_obj[:, :],\n",
+    "        momf_sampler,\n",
+    "        dim_y,\n",
+    "        ref_point,\n",
+    "        standard_bounds,\n",
+    "        BATCH_SIZE,\n",
+    "        cost_call=Cost_callable,\n",
+    "    )\n",
+    "    if verbose:\n",
+    "        print(\"----------------------------\")\n",
+    "        print(\"Iteration Number is \", iteration + 1)\n",
+    "    # Updating train_x and train_obj\n",
+    "    train_x[n_INIT + iteration, :] = new_x\n",
+    "    train_obj[n_INIT + iteration, :] = new_obj\n",
+    "    # reinitialize the models so they are ready for fitting on next iteration\n",
+    "    mll, model = initialize_model(\n",
+    "        train_x[: iteration + 1 + n_INIT, :], train_obj[: iteration + 1 + n_INIT, :]\n",
+    "    )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Result:  Evaluating the Pareto front at the highest fidelity from MOMF\n",
+    "\n",
+    "After the optimization we are interested in evaluating the final Pareto front. For this we train a GP model with the data acquired by the MOMF optimization. After this we generate $10^4$ random test points between between [0,1] with the fidelity dimension set to 1 to take out the Pareto front. Two helper functions are defined to achieve this objective where one function generates the test data and the other extracts the pareto front at the highest fidelity for a given training and testing data."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from botorch.utils.multi_objective.pareto import is_non_dominated\n",
+    "from botorch.utils.multi_objective.hypervolume import Hypervolume\n",
+    "\n",
+    "\n",
+    "def gen_test_points(n_points, dim_x):\n",
+    "    # Function to generate random points with fidelity dimension set to 1. Used to evaluate Pareto front from MOMF\n",
+    "    test_x = torch.rand(size=(n_points, dim_x), **tkwargs)\n",
+    "    test_x[:, -1] = 1\n",
+    "    return test_x\n",
+    "\n",
+    "\n",
+    "def get_pareto(train_x, train_obj, test_x, ref_point):\n",
+    "    # Function that takes in training and testing data with a reference point. It initializes a model and takes out the\n",
+    "    # Posterior mean at the testing points. From these points the non-dominated set is calculated and used to computer\n",
+    "    # the hypervolume.\n",
+    "    mll, model = initialize_model(train_x, train_obj)\n",
+    "    fit_gpytorch_model(mll)\n",
+    "    means = model.posterior(test_x).mean  # Computing posterior mean over outputs at testing data\n",
+    "    pareto_mask = is_non_dominated(means)  # Calculating Non-dominated points\n",
+    "    pareto_f = means[pareto_mask]  # Saving the Pareto front\n",
+    "    temp = Hypervolume(ref_point=torch.tensor(ref_pointMO).to(**tkwargs))  # Used to calculate hypervolume\n",
```

Comment:
let's use `DominatedPartitioning` instead of `Hypervolume` to compute hypervolume. Also, let's use a more informative variable name here